### PR TITLE
feat(nodejs): feature parity [2/N] - add `table.name` and `lancedb.connect({args})`

### PR DESCRIPTION
**Pull request by universalmind303**
depends on https://github.com/lancedb/lancedb/pull/1378

see proper diff here https://github.com/universalmind303/lancedb/compare/remote-table-node...universalmind303:lancedb:table-name

### CODE REVIEW

File: nodejs/lancedb/index.ts [hunk 1]
```diff
@@ -76,15 +71,61 @@ export * as embedding from "./embedding";
...
+ * ```ts
+ * const conn = await connect({
+ *   uri: "/path/to/database",
+ *   storageOptions: {timeout: "60s"}
+ * });
+ * ```
+ */
+export async function connect(
+  opts: Partial<RemoteConnectionOptions | ConnectionOptions> & { uri: string },
+): Promise<Connection>;
```
Comment (wjones127) [inline nodejs/lancedb/index.ts:110]: Should we mark the old style variant with `@deprecated`?

File: nodejs/src/connection.rs [hunk 2]
```diff
@@ -56,12 +56,6 @@ impl Connection {
     #[napi(factory)]
     pub async fn new(uri: String, options: ConnectionOptions) -> napi::Result<Self> {
         let mut builder = ConnectBuilder::new(&uri);
-        if let Some(api_key) = options.api_key {
-            builder = builder.api_key(&api_key);
-        }
-        if let Some(host_override) = options.host_override {
-            builder = builder.host_override(&host_override);
-        }
```
Comment (wjones127) [inline nodejs/src/connection.rs:64]: This was never used, right?

Comment (universalmind303) [reply nodejs/src/connection.rs:64]: yeah. these are now unused as the js side handles the remote connection.

File: nodejs/lancedb/index.ts (hunk 1 above)
Comment (universalmind303) [reply nodejs/lancedb/index.ts:110]: I'm indifferent. I don't think it's much of a hassle supporting both signatures going forward.

Comment (wjones127) [reply nodejs/lancedb/index.ts:110]: Okay that seems like a fine position for now.

**Review by wjones127 — APPROVED** [review]

**Pull request merged**